Add documentation about adding new SIH airframes (#3538)
PX4/PX4-user_guide: en/sim_sih/index.md

make px4_sitl sihsim_quadx
```

### Adding new airframes

> When adding new airframes for use in SIH simulation, there are some differences with respect to ususal airframe files. 

Reviewer: Typo on usual

```suggestion
When adding new airframes for use in SIH simulation, there are some differences with respect to normal airframe files. 
```

For all variants of SIH:

>   - Set the [physical model parameters](../advanced_config/parameter_reference.md#simulation-in-hardware).

Reviewer: You're basically saying set all SIH parameters right? If so

```suggestion
  - Set the all the [Simulation In Hardware](../advanced_config/parameter_reference.md#simulation-in-hardware) parameters (prefixed with `SIH_`) in order to configure the physical model of the vehicle.
```

  - `param set-default SYS_HITL 2` to enable SIH on the next boot.
  - `param set-default CBRK_SUPPLY_CHK 894281` to disable power valid check.
  - `param set-default CBRK_IO_SAFETY 22027` to disable IO safety check.

>   - `param set-default EKF2_GPS_DELAY 0` to improve state estimator performance in the otherwise unrealistic scenario instant GPS measurements. 

Reviewer: Is this what you mean by the comment?

```suggestion
  - `param set-default EKF2_GPS_DELAY 0` to improve state estimator performance (the assumption of instant GPS measurements would normally be unrealistic, but is accurate for SIH).
```

For SIH on FC: 

  - Airframe file goes in `ROMFS/px4fmu_common/init.d/airframes` and follows the naming template `${ID}_${model_name}.hil`, where `ID` is the `SYS_AUTOSTART_ID` used to select the airframe, and `model_name` is the airframe model name.
  - Add the model name in `ROMFS/px4fmu_common/init.d/airframes/CMakeLists.txt` to generate a corresponding make target. 

>   - Actuators are configured with `HIL_ACT_FUNC*`, rather than `PWM_MAIN_FUNC*` as usual. This is to avoid using the actual actuators in SIH. Similarly, the bitfield for inverting individual actuator output ranges is `HIL_ACT_REV`, rather than `PWM_MAIN_REV`.

Reviewer: ```suggestion
  - Actuators are configured with `HIL_ACT_FUNC*`, rather than `PWM_MAIN_FUNC*` as usual.
    This is to avoid using the actual actuators in SIH.
    Similarly, the bitfield for inverting individual actuator output ranges is `HIL_ACT_REV`, rather than `PWM_MAIN_REV`.
```

make px4_sitl sihsim_quadx
```

### Adding new airframes

When adding new airframes for use in SIH simulation, there are some differences with respect to ususal airframe files. 

For all variants of SIH:

  - Set the [physical model parameters](../advanced_config/parameter_reference.md#simulation-in-hardware).

>   - `param set-default SYS_HITL 2` to enable SIH on the next boot.

Reviewer: Comments in airframe says this also stops sensors.

I'm not quite sure how this works w.r.t. sensors. It looks like in the SIH on SITL the sensors must be enabled such as SENS_EN_GPSSIM 1` . On FC I assume that the simulator must take care of providing simulated data for some subset of sensors. Isn't clear if these can be enabled/disabled in the usual way.

  - Add the model name in `ROMFS/px4fmu_common/init.d/airframes/CMakeLists.txt` to generate a corresponding make target. 
  - Actuators are configured with `HIL_ACT_FUNC*`, rather than `PWM_MAIN_FUNC*` as usual.
    This is to avoid using the actual actuators in SIH.
    Similarly, the bitfield for inverting individual actuator output ranges is `HIL_ACT_REV`, rather than `PWM_MAIN_REV`.

For SIH as SITL (no FC):

 - Airframe file goes in `ROMFS/px4fmu_common/init.d-posix/airframes` and follows the naming template `${ID}_sihsim_${model_name}`, where `ID` is the `SYS_AUTOSTART_ID` used to select the airframe, and `model_name` is the airframe model name.
 - Add the model name in `src/modules/simulation/simulator_sih/CMakeLists.txt` to generate a corresponding make target.
 - Actuators are configured as usual with `PWM_MAIN_FUNC*` and `PWM_MAIN_REV`.
 - Additionally, set:
   - `PX4_SIMULATOR=${PX4_SIMULATOR:=sihsim}`
   - `PX4_SIM_MODEL=${PX4_SIM_MODEL:=svtol}` (replacing `svtol` with the airframe model name)
   - `param set-default SENS_EN_GPSSIM 1`
   - `param set-default SENS_EN_BAROSIM 1`

>    - `param set-default SENS_EN_MAGSIM 1`

Reviewer: We might also want to do
```
param set-default SENS_EN_ARSPDSIM 1
```

> 

Reviewer: Perhaps we should link to where they can try specific examples?

```suggestion
For specific examples see the `_sihsim_` airframes in [ROMFS/px4fmu_common/init.d-posix/airframes](https://github.com/PX4/PX4-Autopilot/blob/main/ROMFS/px4fmu_common/init.d-posix/airframes/) (SIH as SITL) and [ROMFS/px4fmu_common/init.d/airframes](https://github.com/PX4/PX4-Autopilot/tree/main/ROMFS/px4fmu_common/init.d/airframes) (SIH on FC).

```

make px4_sitl sihsim_quadx
```

### Adding new airframes

[Adding a new airframe](../dev_airframes/adding_a_new_frame.md) for use in SIH simulation is much the same as for other use cases.

> You still need to configure your vehicle type and [geometry](../config/actuators.md) (`CA_` parameters) and start any other defaults for that vehicle type.

Reviewer 2: ```suggestion
You still need to configure your vehicle type and [geometry](../config/actuators.md) (`CA_` parameters) and start any other defaults for that specific vehicle.
```

The vehicle type in PX4 generally multicopter, VTOL, fixed wing, rover, ... related things should be covered in e.g. https://github.com/PX4/PX4-Autopilot/blob/57fdda597be4e14e3f493f6b55ef550bbb9bdccd/ROMFS/px4fmu_common/init.d/rc.mc_defaults vehicle type I'd only use to refer to that.

The specific differences for SIH simulation airframes are listed in the sections below.

For all variants of SIH:

  - Set all the [Simulation In Hardware](../advanced_config/parameter_reference.md#simulation-in-hardware) parameters (prefixed with `SIH_`) in order to configure the physical model of the vehicle.

>   - `param set-default SYS_HITL 2` to enable SIH on the next boot.

Reviewer: As I understand https://github.com/PX4/PX4-user_guide/pull/3538#issuecomment-2606578011 this is true for both SIH on FC and Simulator. It makes sense to state what the param does so that people know they don't need to start these sensors.

```suggestion
  - `param set-default SYS_HITL 2` to enable SIH on the next boot.
  
     This disables input from real sensors and enables simulated sensor data for [GPS](../advanced_config/parameter_reference.md#simulation-in-SENS_EN_GPSSIM), [barometer](../advanced_config/parameter_reference.md#simulation-in-SENS_EN_BAROSIM ), [magnetometer](../advanced_config/parameter_reference.md#simulation-in-SENS_EN_MAGSIM ) and [Airspeed](../advanced_config/parameter_reference.md#simulation-in-SENS_EN_ARSPDSIM) (used in fixed-wing flight).
```

Author: This is indeed only true for SIH-as-SITL, not for SIH on FC. I did not completely understand that when writing https://github.com/PX4/PX4-user_guide/pull/3538#issuecomment-2606578011, sorry for any confusion.

  - Add the model name in `ROMFS/px4fmu_common/init.d/airframes/CMakeLists.txt` to generate a corresponding make target. 
  - Actuators are configured with `HIL_ACT_FUNC*`, rather than `PWM_MAIN_FUNC*` as usual.
    This is to avoid using the actual actuators in SIH.
    Similarly, the bitfield for inverting individual actuator output ranges is `HIL_ACT_REV`, rather than `PWM_MAIN_REV`.
  - `param set CBRK_USB_CHK 894281` to disable USB link check. 

For SIH as SITL (no FC):

 - Airframe file goes in `ROMFS/px4fmu_common/init.d-posix/airframes` and follows the naming template `${ID}_sihsim_${model_name}`, where `ID` is the `SYS_AUTOSTART_ID` used to select the airframe, and `model_name` is the airframe model name.
 - Add the model name in `src/modules/simulation/simulator_sih/CMakeLists.txt` to generate a corresponding make target.
 - Actuators are configured as usual with `PWM_MAIN_FUNC*` and `PWM_MAIN_REV`.
 - Additionally, set:
   - `PX4_SIMULATOR=${PX4_SIMULATOR:=sihsim}`
   - `PX4_SIM_MODEL=${PX4_SIM_MODEL:=svtol}` (replacing `svtol` with the airframe model name)

>    - `param set-default SENS_EN_GPSSIM 1`
>    - `param set-default SENS_EN_BAROSIM 1`
>    - `param set-default SENS_EN_MAGSIM 1`

Reviewer: ```suggestion
```

Reviewer 2: This comes back to https://github.com/PX4/PX4-Autopilot/commit/5676cc32bc3651ee1f65a918d9d681232c67e995
So all these parameters turn on virtual sensor drivers that take the ground truth data from the simulator (SIH) and publish noisy simulated sensor data to the topics normally used by the PX4 pipeline. As I understand if the parameters are not set to 1 then you don't get any simulated sensor data fed to PX4.
